src/Analyzers/CSharp/Tests/UseNullPropagation/UseNullPropagationTests.cs

```diff
@@ -2069,5 +2069,68 @@ class C
     Action<int> M(List<int> p) => p is null ? null : p.Add;
 }");
         }
+
+        [Fact, WorkItem(66036, "https://github.com/dotnet/roslyn/issues/66036")]
+        public async Task TestElseIfStatement1()
+        {
+            await TestInRegularAndScript1Async("""
+                class C
+                {
+                    void M(string s)
+                    {
+                        if (true)
+                        {
+                        }
+                        else [|if|] (s != null)
+                        {
+                            s.ToString();
+                        }
+                    }
+                }
+                """, """
+                class C
+                {
+                    void M(string s)
+                    {
+                        if (true)
+                        {
+                        }
+                        else
+                        {
+                            s?.ToString();
+                        }
+                    }
+                }
+                """);
+        }
+
+        [Fact, WorkItem(66036, "https://github.com/dotnet/roslyn/issues/66036")]
+        public async Task TestElseIfStatement2()
+        {
+            await TestInRegularAndScript1Async("""
+                class C
+                {
+                    void M(string s)
+                    {
+                        if (true)
+                        {
+                        }
+                        else [|if|] (s != null)
+                            s.ToString();
+                    }
+                }
+                """, """
+                class C
+                {
+                    void M(string s)
+                    {
+                        if (true)
+                        {
+                        }
+                        else s?.ToString();
+                    }
+                }
+                """);
+        }
```

Comment:
Can you add equivalent vb tests to show what the behavior is?

Reply:
There is no even diagnostic for VB. In C#, `else if { ... }` is an `else` clause with `if` statement inside, so the analizer can analize this `if`. In VB, however, the `ElseIf` is a separate syntax, which requires additional work from the analizer to be correctly reported.

Should I still make a test, showing that?

Reply:
Yes.